### PR TITLE
Internal: Move tests into dirs

### PR DESCRIPTION
**Pull request by Subbarker**
## Description

This is because `gotestsum` breaks in [odd ways](https://github.com/gotestyourself/gotestsum/issues/398) when we don't have test directories.

## Related issue

b/330174886

## Checklist:
- Unit tests
  - [x] Unit tests do not apply.
  - [ ] Unit tests have been added/modified and passed for this PR.
- Integration tests
  - [ ] Integration tests do not apply.
  - [x] Integration tests have been added/modified and passed for this PR.
- Documentation
  - [x] This PR introduces no user visible changes.
  - [ ] This PR introduces user visible changes and the corresponding documentation change has been made.
- Minor version bump
  - [x] This PR introduces no new features.
  - [ ] This PR introduces new features, and there is a separate PR to bump the [minor version](https://github.com/GoogleCloudPlatform/ops-agent/blob/master/VERSION) since the last [release](https://github.com/GoogleCloudPlatform/ops-agent/releases) already.
  - [ ] This PR bumps the version.

### CODE REVIEW

File: integration_test/third_party_apps_test/main_test.go [hunk 1]
```diff
@@ -722,14 +722,14 @@ func modifiedFiles(t *testing.T) []string {
 // means we should test all applications.
 func isCriticalFile(f string) bool {
 	if strings.HasPrefix(f, "submodules/") ||
-		strings.HasPrefix(f, "integration_test/third_party_apps_data/agent/") {
+		strings.HasPrefix(f, "integration_test/third_party_apps_test/") {
```
Comment (martijnvans) [inline integration_test/third_party_apps_test/main_test.go:725]: this line is wrong, we should just delete it. The `agent` directory was deleted at some point.

File: integration_test/third_party_apps_test/main_test.go [hunk 2]
```diff
@@ -131,7 +131,7 @@ func distroFolder(platform string) (string, error) {
 }
 
 func readFileFromScriptsDir(scriptPath string) ([]byte, error) {
```
Comment (martijnvans) [inline integration_test/third_party_apps_test/main_test.go:133]: this function is now silly and we should inline its contents into each callsite.

File: kokoro/scripts/test/go_test.sh [hunk 3]
```diff
@@ -208,7 +208,7 @@ ulimit -n 1000000
 
 # Set up some command line flags for "gotestsum".
 gotestsum_args=(
-  --packages=./"${TEST_SUITE_NAME}.go"
+  --packages=./"${TEST_SUITE_NAME}/..."
```
[the remark this answers is not in the document]
Comment (Subbarker) [reply kokoro/scripts/test/go_test.sh:211]: Done

File: integration_test/third_party_apps_test/main_test.go (hunk 1 above)
Comment (Subbarker) [reply integration_test/third_party_apps_test/main_test.go:725]: Done

File: integration_test/third_party_apps_test/main_test.go [hunk 4]
```diff
@@ -722,14 +722,14 @@ func modifiedFiles(t *testing.T) []string {
...
-		strings.HasPrefix(f, "integration_test/third_party_apps_data/agent/") {
+		strings.HasPrefix(f, "integration_test/third_party_apps_test/") {
 		return true
 	}
 	for _, criticalFile := range []string{
 		"go.mod",
 		"integration_test/agents/agents.go",
 		"integration_test/gce/gce_testing.go",
-		"integration_test/third_party_apps_test.go",
+		"integration_test/third_party_apps_test/test.go",
```
[the remark this answers is not in the document]
Comment (Subbarker) [reply integration_test/third_party_apps_test/main_test.go:732]: Done

**Review by martijnvans — APPROVED** [review]

**Pull request merged**